service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapper.java

```diff
@@ -91,6 +104,37 @@ private String prepareComment(Observation observation) {
                     pertinentInformationObservationValueMapper.mapReferenceRangeToPertinentInformation(referenceRange));
             }
         }
+
+        if (observation.hasInterpretation()) {
+            CodeableConcept interpretation = observation.getInterpretation();
+
+            boolean isInterpretationCode = interpretation.getCoding().stream()
+                .anyMatch(this::isInterpretationCode);
+
+            if (!isInterpretationCode) {
+                Optional<Coding> codeWithUserSelected = interpretation.getCoding().stream()
+                    .filter(code -> !isInterpretationCode(code) && code.hasUserSelected() && code.getUserSelected())
+                    .findFirst();
+
+                Coding coding = codeWithUserSelected.orElse(interpretation.getCodingFirstRep());
+
+                if (coding.hasCode() || coding.hasDisplay()) {
+                    if (coding.hasDisplay()) {
+                        commentBuilder.insert(COMMENT_OFFSET, coding.getDisplay() + StringUtils.SPACE);
+                    }
+                    if (coding.hasCode()) {
+                        commentBuilder.insert(COMMENT_OFFSET, coding.getCode() + StringUtils.SPACE);
+                    }
+                    commentBuilder.insert(COMMENT_OFFSET, "Interpretation Code: ");
+                }
```

Comment:
An alternative that avoids double-nesting - your choice
```suggestion
                if (coding.hasDisplay()) {
                    commentBuilder.insert(COMMENT_OFFSET, coding.getDisplay() + StringUtils.SPACE);
                }
                if (coding.hasCode()) {
                    commentBuilder.insert(COMMENT_OFFSET, coding.getCode() + StringUtils.SPACE);
                }
                if (coding.hasCode() || coding.hasDisplay()) {
                    commentBuilder.insert(COMMENT_OFFSET, "Interpretation Code: ");
                }
```

Reply:
Reworked this to avoid double-nesting here 3dbd971